Add examples component to docs (#1729)
gradio-app/gradio: guides/adding_examples_to_your_app.md

# Adding Examples To Your App

Docs: examples

## Introduction

As we saw in the [Quickstart](/getting_started#example-inputs), providing example inputs to your app are a helpful way
to let your audience explore the functionality of your app without having them specify all of your app's expected inputs.

In this guide we'll go into greater depth on how you can provide example inputs to your apps.

## Adding examples to an Interface

As covered in the Quickstart, adding examples to an Interface is as easy as providing a list of lists to the `examples`
keyword argument. 
Each sublist is a data sample, where each element corresponds to an input of the prediction function.

> The inputs must be ordered in the same order as the prediction function expects them.

Reviewer: This is great @freddyaboulton! I would add a couple of things here:
* if the `Interface` has only 1 input component, then you can provide the examples as a regular list instead of a nested sublist
* you can also provide a path to a directory of examples

(Both of these are less common use cases, but I think we should include them in a Guide on examples for completeness).

We can see the `examples` parameter in action in this calculator demo:

$code_calculator
$demo_calculator

## Adding examples to a Blocks app

In order to add examples to an app written with the Blocks API, you need to use the `gr.Examples` component helper.
This class is a wrapper over the `gr.Dataset` component.

During class creation, `gr.Examples` will instantiate a `gr.Dataset` and attach an event listener on the dataset click event
that will populate all the inputs with the values of that row of the dataset. 

To show this in practice, we will build the same calculator but with Blocks instead of an Interface.

$code_calculator_blocks
$demo_calculator_blocks

By the way, Interface uses `gr.Examples` under the hood too!
So if you know how this example works you also know how the Interface examples work! 🥳

## Caching examples

Both `gr.Examples` and `Interface` have a `cache_examples` parameter.
The default in the previous demos is to not cache examples, but if `cache_examples=True`, the `gr.Examples` component will

> run all of your examples through your app and save the outputs during start up.

Reviewer: ```suggestion
run all of your examples through your app and save the outputs when you call the `launch()` method.
```

> Whenever someone clicks on an example, the output will automatically be populated in the app. 

Reviewer: ```suggestion
Whenever someone clicks on an example, the output will automatically be populated in the app. This is useful especially when your model can take a while to run! 
```
